source/fundamentals/crud/read-operations/retrieve.txt

```diff
@@ -0,0 +1,170 @@
+To follow along with the examples on this page, run the following
+snippet to load the documents into the ``ratings`` collection in the
+``tea`` database:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/retrieve.go
+   :language: go
+   :dedent:
+   :start-after: begin insert docs
+   :end-before: end insert docs
+
+Each document contains the tea a person tried and their rating on a
+scale of 10, which corresponds to the ``name`` and ``rating`` fields.
+
+.. note::
+
+   Each example truncates the ``ObjectID`` value since the driver
+   generates them uniquely. 
+
+.. _retrieve-find:
+
+Find operations
+---------------
+
+Use a **find operation** to retrieve a subset of your existing data.
+Find operations consist of the ``Find()`` and ``FindOne()`` functions.
+
+Find All Documents
+~~~~~~~~~~~~~~~~~~
+
+The ``Find()`` function expects you to pass a ``Context`` type and a
+filter. The function returns *all* documents that match the filter as a
+``Cursor`` type. If you pass an empty filter, the function returns *all*
+documents in the collection. 
+
+.. To learn how to access data in a cursor, see the :doc:`Cursor
+.. </fundamentals/crud/read-operations/cursor>` guide. 
+
+Find One Document
+~~~~~~~~~~~~~~~~~
+
+The ``FindOne()`` function expects you to pass a ``Context`` type and a
+filter. The function returns *the first document* that matches the
+filter as a ``SingleResult`` type. If you pass an empty filter, the
+function returns *a random document* in the collection.
+
+To learn how to access data in a ``SingleResult`` see the :ref:`BSON
+<bson-unmarshalling>` guide.
+
+Example
+```````
+
+The following example passes a context and a filter to the ``Find()``
+function, which matches documents where the ``ratings`` field falls
+between ``5`` and ``10``:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/retrieve.go
+   :language: go
+   :dedent:
+   :start-after: begin find docs
+   :end-before: end find docs
+
+After running the preceding example, the output resembles the following:
+
+.. code-block:: none
+   :copyable: false
+
+    //results truncated
+   [{_id ObjectID("...")} {name Chai} {rating 7}]
+   [{_id ObjectID("...")} {name Earl Grey} {rating 9}]
+
+For a runnable ``find()`` example, see the :doc:`Find Multiple
+Documents </usage-examples/find>` page. 
+
+.. _retrieve-aggregate:
+
+Aggregate Operation
+-------------------
+
+Use the **aggregate operation** to retrieve and transform a subset of
+your existing data. The aggregate operation consists of the
+``Aggregate()`` function.
+
+Aggregation
+~~~~~~~~~~~
+
+The ``Aggregate()`` function expects you to pass a ``Context`` type and
+an **aggregation pipeline**. An aggregation pipeline defines how to
+transform data through stages such as specifying how to match documents,
+rename fields, and group values. 
+
+The function returns the resulting documents in a ``Cursor`` type. If
+you omit the ``match`` stage, the pipeline proceeds using all documents
```

Comment:
nit and suggestion: Since this is a one-off reference to the `$match` stage (which is what it is called), consider linking to the `$match` documentation.

```suggestion
you omit the :ref:`$match <some-server-ref>` stage, the pipeline proceeds using all documents
```

Reply:
there wasn't a ref on [their page](https://github.com/mongodb/docs/blob/master/source/reference/operator/aggregation/match.txt), so I used the snooty directive